File: server/datasets/java_source-code/Multiple-F1/mixed_attack_36_merge_sort.java
import java.util.Arrays;

public class Solution {
    public int[] mergeSort(int[] nums) {
        if (nums == null || nums.length <= 1) return nums;
        int mid = nums.length / 2;
        int[] left = mergeSort(Arrays.copyOfRange(nums, 0, mid));
        int[] right = mergeSort(Arrays.copyOfRange(nums, mid, nums.length));
        
        int i = 0, j = 0, k = 0;
        while (i < left.length && j < right.length) {
            if (left[i] < right[j]) nums[k++] = left[i++];
            else nums[k++] = right[j++];
        }
        for (int p = i; p < left.length; p++) nums[k++] = left[p];
        for (int q = j; q < right.length; q++) nums[k++] = right[q];
        return nums;
    }
}

File: server/datasets/python_source-code/Multiple-F1/mixed_attack_36_merge_sort.py
def merge_sort(values_arr):
    if len(values_arr) <= 1:
        return values_arr
    mid = len(values_arr) // 2
    left = merge_sort(values_arr[:mid])
    right = merge_sort(values_arr[mid:])
    
    i = j = k = 0
    while i < len(left) and j < len(right):
        if left[i] < right[j]:
            values_arr[k] = left[i]
            i += 1
        else:
            values_arr[k] = right[j]
            j += 1
        k += 1
    for p in range(i, len(left)):
        values_arr[k] = left[p]
        k += 1
    for q in range(j, len(right)):
        values_arr[k] = right[q]
        k += 1
    return values_arr
